### DISCUSSION TITLE
TCP+TLS+VLESS、HTTPS、WSS，三种连接方式在流量特征上有区别吗？

### DISCUSSION THREAD

**Discussion by lanchengjiu** [XTLS/Xray-core#605] [Q&A]
我是这样理解的，HTTPS和WSS都是基于TCP的，层级关系是最外层TCP、中层TLS、内层VLESS/HTTP/WS
TLS加密可以认为是绝对安全的，那么从第三方看，这三种连接都是TCP+TLS，内层具体是什么看不出区别，不知道我理解的对不？

从主动探测的角度考虑，可以试探下服务器是不是一个WEB，不过也就仅此而已了，因为TCP只暴露IP和端口，路径等都在TLS里面
那么进一步在思考，XTLS因为一会是正规网站的HTTPS，一会是代理的TLS，证书并不一致，TLS握手的过程中会不会暴露什么特征呢？

**Comment by RPRX** [comment]
对错杂糅，一言难尽。

**Reply by galaxyskyknight** [reply]
为什么证书会不一致呢？我用回落nginx反代其他正规网站，但nginx的反代配置里ssl_certificate就指向本地的xray的证书路径，对外访问证书不管是xray的xtls/tls还是回落网站都是一样的证书。
